### DISCUSSION TITLE
Passing a URL variable to a query

### DISCUSSION THREAD

**Discussion by PocydonZach** [Budibase/budibase#6997] [Help]
I am using a custom query as the main data provider for a screen. What are the steps to pass a URL variable to this query for use as an SQL statement variable?

Also, is is possible to also use this URL variable in the query transformer?

**Comment by DennisNissenUX** [comment]
Hey @PocydonZach 
I have done a similar thing last week and it worked very well for me.

First I would have a look in the budibase docs:
- https://docs.budibase.com/docs/rest-queries
- https://docs.budibase.com/docs/rest-bindings

I found the tutorial to be somewhat helpful for me.

A few extra comments:
- in the action where you call the query you can also select parameters from the URL as binding value. For me, this was displayed directly in the "bindable values"
- in the settings of your query you can create bindings. you can use them via handlebar statements (e.g. {{example}} )in the URL but also in the query transformer
- it can be a bit tricky but with some try and error you will get it to work

I hope this is helpful.

**Reply by shogunpurple** [reply]
This is the solution ^ thank you @DennisNissenUX - really appreciate the feedback. I've just created a ticket to make this easier and we will implement a better UX around it https://github.com/Budibase/budibase/issues/7014

Do you have anything in particular you would like to be easier here? My theory was that the problem is screen creation, as you must set up your screen in a certain way in order to take advantage of URL params.

**Reply by DennisNissenUX** [reply]
Thanks 👍 .
I have written a small comment in the issue with an idea that could make the process a bit more accessible.
